bukkit/src/main/java/me/lokka30/treasury/plugin/bukkit/BukkitTreasuryPlugin.java

```diff
@@ -0,0 +1,255 @@
+        ServicePriority priority = lowPriority ? ServicePriority.Low : ServicePriority.High;
+        if (provider.getPriority().equalsIgnoreCase(priority.name().toLowerCase(Locale.ROOT))) {
+            return;
+        }
+        Plugin plugin = Bukkit.getPluginManager().getPlugin(provider.registrar().getName());
+        EconomyProvider eco = provider.provide();
+        Bukkit.getServicesManager().unregister(eco);
+        Bukkit.getServicesManager().register(EconomyProvider.class, eco, plugin, priority);
+    }
+
+    @Override
+    public void unregisterProvider(@NotNull EconomyProvider provider) {
+        runSync(() -> Bukkit.getServicesManager().unregister(provider));
+    }
+
+    @Override
+    public @NotNull Logger logger() {
+        return this;
+    }
+
+    @Override
+    public @NotNull Scheduler scheduler() {
+        return this;
+    }
+
+    @Override
+    public @NotNull ConfigAdapter configAdapter() {
+        return this;
+    }
+
+    @Override
+    public void reload() {
+        messages = new MessagesImpl(plugin);
+        messages.load();
+        this.loadSettings(false);
+    }
+
+    public void loadSettings(boolean regenerate) {
+        File settingsFile = new File(plugin.getDataFolder(), "settings.yml");
+        if (regenerate && settingsFile.exists()) {
+            // todo: this is not the correct way of dumping new options. find another way!!!!
+            settingsFile.delete();
+        }
+        if (!settingsFile.exists()) {
+            if (!settingsFile.getParentFile().exists()) {
+                settingsFile.getParentFile().mkdirs();
+            }
+            try (InputStream in = getClass().getClassLoader().getResourceAsStream("settings.yml")) {
+                Files.copy(in, settingsFile.getAbsoluteFile().toPath());
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        settings = YamlConfiguration.loadConfiguration(settingsFile);
+    }
+
+    @Override
+    public @NotNull EconomyAPIVersion getEconomyAPIVersion() {
+        return Treasury.ECONOMY_API_VERSION;
+    }
+
+    @Override
+    public @NotNull List<String> pluginsList() {
+        if (cachedPluginList != null) {
+            return cachedPluginList;
+        }
+        cachedPluginList = Arrays.stream(
+                Bukkit.getPluginManager().getPlugins()
+        ).map(Plugin::getName).collect(Collectors.toList());
+        return cachedPluginList;
+    }
+
+    @Override
+    public @NotNull Messages getMessages() {
+        return messages;
+    }
+
+    @Override
+    public @NotNull Settings getSettings() {
+        return this;
+    }
+
+    public String colorize(@NotNull String message) {
+        return fork.isSpigot()
+                ? net.md_5.bungee.api.ChatColor.translateAlternateColorCodes('&', message)
+                : org.bukkit.ChatColor.translateAlternateColorCodes('&', message);
```

Comment:
Is there a real difference between these two?

~~also, ew, legacy color codes~~

Reply:
yes. BungeeCord's ChatColor supports hex colors, while bukkit's doesn't.